base/strings/util.jl

```diff
@@ -248,6 +248,123 @@ julia> rpad("March",20)
 rpad(s, n::Integer, p=" ") = rpad(string(s),n,string(p))
 cpad(s, n::Integer, p=" ") = rpad(lpad(s,div(n+strwidth(s),2),p),n,p)
 
+immutable SplitIterator
+    str::AbstractString
+    splitter
```

Comment:
This phenomena also seems to occur with the `graphemes` vs. `split(str, "")` which do nearly similar things. I've tried to maintain the invariants between the iterator version and the `split` version. We'll see which invariants I've failed to satisfy when I'm able to debug these tests. :)